server/src/main/java/org/elasticsearch/indices/TimestampFieldMapperService.java

```diff
@@ -176,7 +178,17 @@ private static boolean hasUsefulTimestampField(IndexMetadata indexMetadata) {
         if (future == null || future.isDone() == false) {
             return null;
         }
-        return future.actionGet();
+
+        try {
+            // It's possible that callers of this class are executed
+            // in a transport thread, for that reason we request
+            // the future value with a timeout of 0. That won't
+            // trigger assertion errors.
+            return future.actionGet(TimeValue.ZERO);
+        } catch (ElasticsearchTimeoutException e) {
+            assert false : "Unexpected timeout exception while getting a timestamp mapping";
+        }
```

Comment:
Let's not fail the whole node if we messed this up somehow :)

```suggestion
            throw e;
        }
```